ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/ccd/repositories/CCDCaseApi.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.hmcts.cmc.ccd.migration.ccd.repositories;
+
+import com.google.common.collect.ImmutableMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.ccd.migration.idam.models.User;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+@ConditionalOnProperty(prefix = "core_case_data", name = "api.url")
+public class CCDCaseApi {
+
+    public static final String JURISDICTION_ID = "CMC";
+    public static final String CASE_TYPE_ID = "MoneyClaimCase";
+
+    private final CoreCaseDataApi coreCaseDataApi;
+    private final AuthTokenGenerator authTokenGenerator;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CCDCaseApi.class);
+
+    public CCDCaseApi(
+        CoreCaseDataApi coreCaseDataApi,
+        AuthTokenGenerator authTokenGenerator
+    ) {
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.authTokenGenerator = authTokenGenerator;
+    }
+
+    public Optional<Long> claimExists(User user, String referenceNumber) {
+        LOGGER.info("Get claim from CCD " + referenceNumber);
+
+        Optional<Long> ccdId = search(user, ImmutableMap.of("case.referenceNumber", referenceNumber));
+
+        if (ccdId.isPresent()) {
+            LOGGER.info("Claim found " + ccdId.get());
+        }
+
+        return ccdId;
+    }
+
+    private Optional<Long> search(User user, Map<String, Object> searchString) {
+
+        String serviceAuthToken = this.authTokenGenerator.generate();
+
+        List<CaseDetails> result;
+        result = this.coreCaseDataApi.searchForCaseworker(
+            user.getAuthorisation(),
+            serviceAuthToken,
+            user.getUserDetails().getId(),
+            JURISDICTION_ID,
+            CASE_TYPE_ID,
+            searchString
+        );
+
+        return result.size() > 0 ? Optional.of(result.get(0).getId()) : Optional.empty();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS1155)